crud/select.lua

```diff
@@ -222,31 +198,32 @@ function select_module.pairs(space_name, user_conditions, opts)
         error(string.format("Failed to generate iterator: %s", err))
     end
 
-    local gen = function(_, iter)
-        if not iter:has_next() then
+    local tuple, _
+
+    if opts.use_tomap ~= true then
+        return iter.iter:pairs()
+    end
+
+    local merger_gen = iter.iter:pairs()
+    local gen = function()
+        _, tuple = merger_gen.gen(nil, merger_gen.state)
+        if tuple == nil then
             return nil
         end
 
-        local tuple, err = iter:get()
+        local result
+        result, err = utils.unflatten(tuple, iter.space_format)
         if err ~= nil then
-            error(string.format("Failed to get next object: %s", err))
-        end
-
-        local result = tuple
-        if opts.use_tomap == true then
-            result, err = utils.unflatten(tuple, iter.space_format)
-            if err ~= nil then
-                error(string.format("Failed to unflatten next object: %s", err))
-            end
+            error(string.format("Failed to unflatten next object: %s", err))
         end
 
         return iter, result
     end
 
-    return gen, nil, iter
+    return gen
 end
 
-function select_module.call(space_name, user_conditions, opts)
+local function select_module_call_xc(space_name, user_conditions, opts)
```

Comment:
What does `_xc` mean?

Reply:
https://github.com/tarantool/tarantool/search?q=xc

Reply:
Ok, but it's not hte only one function that can raise in crud, but the only one with `_xc` suffix.

Reply:
- This is not a public function
- Other functions returns support `nil, err` contract.